x-pack/plugins/security_solution/public/management/pages/trusted_apps/view/components/effected_policy_select/effected_policy_select.tsx

```diff
@@ -0,0 +1,151 @@
+    ...otherSelectableProps
+  }) => {
+    const DEFAULT_LIST_PROPS = useMemo(() => ({ bordered: true }), []);
+
+    const [selectionState, setSelectionState] = useState<EffectedPolicySelection>({
+      isGlobal,
+      selected,
+    });
+
+    const getTestId = useCallback(
+      (suffix): string | undefined => {
+        if (dataTestSubj) {
+          return `${dataTestSubj}-${suffix}`;
+        }
+      },
+      [dataTestSubj]
+    );
+
+    const selectableOptions: EffectedPolicyOption[] = useMemo(() => {
+      const isPolicySelected = selected.reduce<{ [K: string]: PolicyData }>(
+        (idToPolicyData, policy) => {
+          idToPolicyData[policy.id] = policy;
+          return idToPolicyData;
+        },
+        {}
+      );
+
+      return options
+        .map<EffectedPolicyOption>((policy) => ({
+          label: policy.name,
+          policy,
+          checked: isPolicySelected[policy.id] ? 'on' : undefined,
+          disabled: isGlobal,
+        }))
+        .sort(({ label: labelA }, { label: labelB }) => labelA.localeCompare(labelB));
+    }, [isGlobal, options, selected]);
+
+    const handleOnPolicySelectChange = useCallback<
+      Required<EuiSelectableProps<OptionPolicyData>>['onChange']
+    >((currentOptions) => {
+      setSelectionState((prevState) => ({
+        ...prevState,
+        selected: currentOptions.filter((opt) => opt.checked).map((opt) => opt.policy),
+      }));
+    }, [])!;
+
+    const handleGlobalSwitchChange: EuiSwitchProps['onChange'] = useCallback(
+      ({ target: { checked } }) => {
+        setSelectionState((prevState) => ({ ...prevState, isGlobal: checked }));
+      },
+      []
+    );
+
+    const listBuilderCallback: EuiSelectableProps['children'] = useCallback((list) => list, []);
+
+    // Anytime selection state is updated, call `onChange`
+    useEffect(() => {
+      onChange(selectionState);
+    }, [onChange, selectionState]);
+
+    return (
+      <>
+        <EuiFormRow
+          fullWidth
+          label={i18n.translate(
+            'xpack.securitySolution.trustedapps.policySelect.globalSectionTitle',
+            {
+              defaultMessage: 'Global application',
+            }
+          )}
+        >
+          <EuiSwitch
+            label={i18n.translate(
+              'xpack.securitySolution.trustedapps.policySelect.globalSwitchTitle',
+              {
+                defaultMessage: 'Apply trusted application globally',
+              }
+            )}
+            checked={selectionState.isGlobal}
+            onChange={handleGlobalSwitchChange}
+            data-test-subj={getTestId('globalSwitch')}
+          />
+        </EuiFormRow>
+        <EuiFormRow
+          fullWidth
+          label={i18n.translate('xpack.securitySolution.policySelect.policySpecificSectionTitle', {
+            defaultMessage: 'Apply to specific endpoint policies',
+          })}
+        >
+          <EuiSelectable<OptionPolicyData>
+            {...otherSelectableProps}
```

Comment:
This is nicely flexible, somehow I'm still leaning towards more explicit property passing, gives better encapsulation properties and makes it more clear what author tries to control in this component.

Reply:
EuiSelectable is supper flexible and I did not want to take away from that, thus why I normally just re-expose a component's props when I "extend" it. I can remove it and only explicitely expose what the component requires if you feel strongly about it.
Let me know

Reply:
Definitely not strongly, leaving it up to you.